mubench.pipeline/tasks/implementations/collect_misuses.py

```diff
@@ -0,0 +1,10 @@
+from data.project_version import ProjectVersion
+from utils.data_filter import DataFilter
+
+
+class CollectMisuses:
+    def __init__(self, data_filter: DataFilter):
+        self.data_filter = data_filter
+
+    def run(self, version: ProjectVersion):
+        return [misuse for misuse in version.misuses if not self.data_filter.is_filtered(misuse.id)]
```

Comment:
We do when publishing metadata. There's probably a reasonable way around this I guess. It would definitely be nice to get some of that logic out of the data classes.

Reply:
It's actually also used quite extensively by the `CompileTask` (to access patterns and the misuse source files). Unfortunately, there's no way for a task to request to entire list of values as a parameter and I don't think we can support this, as type hints cannot specify the element type of a `List`... we could replace the usages of `version.misuses` by `CollectMisusesTask().run(version)`... though that's ugly and we'd need the filter for it to work, right?... ideas?